src/AppInstallerCLICore/Workflows/MSStoreInstallerHandler.cpp

```diff
@@ -212,72 +317,71 @@ namespace AppInstaller::CLI::Workflow
     void MSStoreUpdate(Execution::Context& context)
     {
         auto productId = Utility::ConvertToUTF16(context.Get<Execution::Data::Installer>()->ProductId);
-
-        // Best effort verifying/acquiring product ownership.
-        std::ignore = EnsureFreeEntitlement(context, productId);
-
-        AppInstallManager installManager;
-        AppUpdateOptions updateOptions;
-
-        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
-
-        // SearchForUpdateAsync will automatically trigger update if found.
-        AppInstallItem installItem = installManager.SearchForUpdatesAsync(
-            productId,          // ProductId
-            winrt::hstring(),   // SkuId
-            winrt::hstring(),
-            winrt::hstring(),   // ClientId
-            updateOptions
-        ).get();
-
-        if (!installItem)
+        auto scope = Manifest::ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope));
+        HRESULT hr = MSStoreUpdateImpl(context, productId, scope, false);
+        if (FAILED(hr))
         {
-            context.Reporter.Info() << Resource::String::UpdateNotApplicable << std::endl
-                << Resource::String::UpdateNotApplicableReason << std::endl;
-            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE);
+            AICLI_TERMINATE_CONTEXT(hr);
         }
+    }
 
-        std::vector<AppInstallItem> installItemVector{ installItem };
-        IVectorView<AppInstallItem> installItems = winrt::single_threaded_vector(std::move(installItemVector)).GetView();
-
-        HRESULT errorCode = WaitForMSStoreOperation(context, installItems);
+    void EnsureStorePolicySatisfied(Execution::Context& context)
+    {
+        auto productId = Utility::ConvertToUTF16(context.Get<Execution::Data::Installer>()->ProductId);
+        HRESULT hr = EnsureStorePolicySatisfiedImpl(context, productId);
+        if (FAILED(hr))
+        {
+            AICLI_TERMINATE_CONTEXT(hr);
+        }
+    }
 
-        if (SUCCEEDED(errorCode))
+    void EnableConfiguration(Execution::Context& context)
+    {
+#ifndef AICLI_DISABLE_TEST_HOOKS
+        HRESULT hr = AppInstallerUpdate(context, false);
+        if (FAILED(hr))
         {
-            context.Reporter.Info() << Resource::String::InstallFlowInstallSuccess << std::endl;
+            AICLI_TERMINATE_CONTEXT(hr);
+        }
+#else
+        if (IsStubPackage())
+        {
+            context.Reporter.Info() << Resource::String::ConfigurationEnablingMessage << std::endl;
+            HRESULT hr = AppInstallerUpdate(context, false);

```

Comment:
What is the value of returning an `HRESULT` if you are just going to terminate the context if it failed?  Why are you converting these away from workflow tasks?

My suggestion was to convert the entire thing into back end code and then the front end would need to be written to watch it.  I don't think that must be done here (although I don't think I like the messages being output in the stub updates), but I do think it should either remain workflow tasks (`void F(Context&)`) or be completely separated (taking no `Context` at all and implemented in CommonCore).